### ISSUE TITLE
DocumentNode types dont match between graphql-tag 12.2.6 and graphql 16.6.0 for typescript projects

### ISSUE THREAD

**Issue by steelerc69** [apollographql/graphql-tag#659]
Error message received during compile is:

    Types of property 'kind' are incompatible.
      Type '"Document"' is not assignable to type 'Kind.DOCUMENT'.

**Comment by steelerc69** [comment]
Seems to be a node resolution issue! not yours.  sorry!

**Comment by josehernandezv** [comment]
How did you fix it @steelerc69 ? I'm having the same issue

**Comment by smcgivern** [comment]
In our case, we had a combination of factors:

1. `graphql-tag` was in `devDependencies`.
2. `graphql` was in `dependencies`.
3. A library that had a dependency on a different version of `graphql`, along with a dependency on `graphql-tag`, was also in `dependencies`.
4. We installed dependencies with `pnpm install --prod --shamefully-hoist`, which meant we got the peer versions in 3 above, not the version from 1.
